src/osd/OSD.cc

```diff
@@ -3459,6 +3460,20 @@ void OSD::_add_heartbeat_peer(int p)
     HeartbeatSession *s = new HeartbeatSession(p);
     hi->con_back = cons.first.get();
     hi->con_back->set_priv(s->get());
+    map<string, string> reporter_loc = osdmap->crush->get_full_location(p);
+    map<string, string>::iterator iter = reporter_loc.find(heartbeat_peers_subtree_level);
+    string st;
+    if (iter == reporter_loc.end()) {
+      st = "osd." + to_string(p);
+    } else {
+      st = iter->second;
+    }
+    if (heartbeat_peers_by_subtree.count(st)) {
+      ++heartbeat_peers_by_subtree[st];
+    } 
```

Comment:
just this increment line is sufficient; if the vaule isn't there yet it'll get initialized to 0 (then incremented).